### PR TITLE
Feature display talents

### PR DESCRIPTION
**Pull request by SheldonCheng**
- Expand existing jobs fetch and populate to work for both jobs and talents.
- Add category selector

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ function App() {
       <Layout>
         <Switch>
           <Route exact path="/" component={HomeScreen} />
-          <Route path="/posts" component={PostsScreen} />
+          <Route path="/:category(jobs|talents)/:id?" component={PostsScreen} />
```
Comment (subhransu) [inline ui/src/App.tsx]: what will the `talents` route do?

File: ui/src/App.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ function App() {
...
           <Route path="/favourite" component={FavouriteScreen} />
```
Comment (subhransu) [inline ui/src/App.tsx]: should we use `favourites` to keep it consistent with `jobs`

File: ui/src/screens/Posts/hooks/usePostsReducer.ts [hunk 3]
```diff
@@ -85,47 +106,53 @@ function PostsReducer(state: PostsState, action: PostsActionTypes): PostsState {
...
-  const match = useRouteMatch<{ id: string }>("/posts/:id");
-  const id = match?.params?.id;
+  const { category: categoryString, id } = useParams<{ category: string; id: string }>();
+  const category = categoryString as CategoryTypes;
 
   React.useEffect(() => {
-    axios.get<JobsResponseData>("/api/jobs").then((res) => {
-      updateJobs(res.data.jobs);
-      if (id) setActiveJob(id);
+    axios.get(`/api/${category}`).then((res) => {
```
Comment (SheldonCheng) [inline ui/src/screens/Posts/hooks/usePostsReducer.ts]: @lxcid I had some trouble with this `axios.get` part.

You earlier added the jobs response interface: `axios.get<JobsResponseData>(...)`

But now, depending on category, the response will either be
`{ jobs: PostInterface[] }`
or
`{ talents: PostInterface[] }`

I can't figure out how to do this, or if it even makes sense, so I removed it for now.

File: ui/src/App.tsx (hunk 1 above)
Comment (lxcid) [reply ui/src/App.tsx]: Should we split it to `/talents` and manage the talents in a separate `TalentsScreen`?

Comment (subhransu) [reply ui/src/App.tsx]: ahh, so `talents` are people looking for jobs? If so it's fine to have it as it is as the UI is not going to change much.

**Comment by subhransu** [comment]
@lxcid help to review this as I am unable to understand much of the ReactJS code.

File: ui/src/screens/Posts/hooks/usePostsReducer.ts (hunk 3 above)
Comment (lxcid) [reply ui/src/screens/Posts/hooks/usePostsReducer.ts]: noted! I think I can figure that out later. :)

**Comment by lxcid** [comment]
ok! I have a PR with some changes for @SheldonCheng to review that merge to this branch though.

https://github.com/SheldonCheng/jobsgowhere/pull/1

**Review by lxcid — APPROVED** [review]
LGTM!

**Comment by subhransu** [comment]
@SheldonCheng rebase and merge.

**Comment by lxcid** [comment]
I'm gonna rebase and merge through GitHub instead. Gonna get this in to work on new branch. :)

**Pull request merged**